enigma-core/app/src/networking/ipc_listener.rs

```diff
@@ -230,6 +233,28 @@ pub(self) mod handling {
         Ok(IpcResponse::UpdateNewContract { address, result: IpcResults::Status(0) })
     }
 
+    #[logfn(INFO)]
+    pub fn update_new_contract_on_deployment(db: &mut DB, address: String, bytecode: &str, delta: IpcDelta) -> ResponseResult {
+        let mut tuples = Vec::with_capacity(DEPLOYMENT_VALS_LEN);
+        let address_arr = ContractAddress::from_hex(&address)?;
+
+        let bytecode = bytecode.from_hex()?;
+        let bytecode_delta_key = DeltaKey::new(address_arr, Stype::ByteCode);
+        tuples.push((bytecode_delta_key, bytecode));
+
+        let data = delta.data.ok_or(P2PErr { cmd: "UpdateNewContractOnDeployment".to_string(), msg: "Delta Data Missing".to_string() })?;
+        let delta_key = DeltaKey::new(address_arr, Stype::Delta(delta.key));
+        tuples.push((delta_key, data));
+
+        let results = db.insert_tuples(&tuples);
+        let mut status: i8 = 0;
+        for result in results {
```

Comment:
It is better to replace -1 with FAILED. In addition, IMO use of `any` would be nicer here. Something like that:
if results.into_iter().any(|&x| x.is_err()){
      status = FAILED;
}